src/object.ts

```diff
@@ -275,17 +290,24 @@ export class KubernetesObjectApi extends ApisApi {
      * @param options Optional headers to use in the request.
      * @return Promise containing the request response and [[KubernetesObject]].
      */
-    public async read(
-        spec: KubernetesObject,
+    public async read<T extends KubernetesObject | KubernetesObject>(
+        spec: KubernetesObjectHeader<T>,
         pretty?: string,
         exact?: boolean,
         exportt?: boolean,
         options: { headers: { [name: string]: string } } = { headers: {} },
-    ): Promise<{ body: KubernetesObject; response: http.IncomingMessage }> {
+    ): Promise<{ body: T; response: http.IncomingMessage }> {
         // verify required parameter 'spec' is not null or undefined
         if (spec === null || spec === undefined) {
             throw new Error('Required parameter spec was null or undefined when calling read.');
         }
+        // verify required parameter 'kind' is not null or undefined
+        if (spec.kind === null || spec.kind === undefined) {
+            throw new Error('Required parameter spec.kind was null or undefined when calling read.');
+        }
+        if (!spec.apiVersion) {
```

Comment:
I think we should throw an Error here instead of defaulting, I suspect that this will confuse people.

Reply:
Done